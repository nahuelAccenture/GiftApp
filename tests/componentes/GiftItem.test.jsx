import { render, screen } from "@testing-library/react";
import { GiftItem } from "../../src/components/GiftItem";

describe('Pruebas en GridItem', () => {

    const title = "Sitama";
    const url ="https://one-punchh/saitma.jpg";

    test('Debe hacer match con el snapshot', () => {
       const {container} = render(<GiftItem title={title} url={url}/>)
        expect(container).toMatchSnapshot();
    });

    test('debe mostrar la url con la imagen y el ALT indicado', () => {
        render(<GiftItem title={title} url={url}/>);
        // screen debug.
        // expect( screen.getByRole('img').src).toBe(url);
        // expect( screen.getByRole('img').alt).toBe(title);
        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(alt);

    });

    test('debe mostrar el tÃ­tulo en el componente', () => {
        render( <GiftItem title={title} url={url}/>);
        expect(screen.getByText(title)).toBeTruthy();
    })
});

