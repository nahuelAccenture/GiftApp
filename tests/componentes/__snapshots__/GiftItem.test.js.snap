// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GridItem Deberia mostrar un error si no aparece el titulo 1`] = `undefined`;
